kstreams/middleware/middleware.py

```diff
@@ -0,0 +1,70 @@
+import logging
+import typing
+
+from aiokafka import errors
+
+from kstreams import ConsumerRecord, types
+
+if typing.TYPE_CHECKING:
+    from kstreams import Stream
+
+
+logger = logging.getLogger(__name__)
+
+
+class Middleware(typing.Protocol):
+    def __init__(
+        self,
+        *,
+        next_call: types.NextMiddlewareCall,
+        send: types.Send,
+        stream: "Stream",
+        **options: typing.Any,
+    ) -> None:
+        ...
+
+    async def __call__(self, cr: ConsumerRecord) -> None:
+        ...
+
+
+class MiddlewareFactory:
+    def __init__(
+        self, middleware: typing.Type[Middleware], **options: typing.Any
+    ) -> None:
+        self.middleware = middleware
+        self.options = options
```

Comment:
What are the options here used for?

Reply:
It is to supply extra arguments to a `middleware` in case that is needed. Perhaps this will clarify it: https://kpn.github.io/kstreams/pr-preview/pr-155/middleware/#adding-options-to-the-middleware